app/views/buyers/applications/_listing.html.slim

```diff
@@ -18,7 +18,7 @@ div id="bulk-operations" style="display: none;"
     = javascript_include_tag 'bulk_operations.js'
     = javascript_include_tag 'vendor/jquery.tipsy.js'
 
-- show_application_plans = @stock_and_custom_application_plans > 1 && !master_on_premises?
+- show_application_plans = @application_plans.size > 1 && !master_on_premises?
```

Comment:
https://github.com/3scale/porta/pull/1798/files#r404884739

Now I see why the `before_filter`. Maybe rename the method in the controller then.

Reply:
Like it was before, `find_application_plans`? I also added an alias to make it look cleaner 🦐

Reply:
Personally, I like to use the `before_filter` when at least one of the the following conditions apply:
1. more than one action depends on what it does;
2. it can abort the request, raising an immediate response (e.g. 404, 403, etc) before getting into the details of the specific action requested, i.e. before starting the actual transaction that could cause the state of the application to mutate. Not necessarily because the payload is invalid (this would be too much for a controller to decide), but because the request can be rejected right away.

I think your `find_accessible_plans` (aliases `accessible_plans`) is somehow in the middle. It doesn't seem to fit any of the conditions above entirely. It won't raise a 404 or 403 in any case, so condition `#2` is completely out. But even condition `#1` is not fully satisfied. The instance variable `@application_plans` is only needed for the `#index` action, while other actions do need to fetch "accessible plans" as well but without setting the instance variable.

I see no action referring to `accessible_plans` more than once. So this is what I would do. I would remove the before filter and the memoization. I'd make the `#index` action to set `@applications_plans = accessible_plans` and then use `@applications_plans` instead of `accessible_plans` further down. As for the other actions, just keep `accessible_plans` as is.